### DISCUSSION TITLE
Feature idea: easy export to an established metabolomics repository

### DISCUSSION THREAD

**Discussion by mneinast** [Princeton-LSI-ResearchComputing/tracebase#1375] [Ideas]
_(right now, I don't see this as a feature to implement before publishing tracebase)..._

When we publish a manuscript, we are supposed to upload our raw files / data to a repository.  Long term, we could use a public-facing instance of TraceBase for this, where the focus on a limited type of experiment (isotope tracing) provides benefits.

However, there are already a couple of repositories for general metabolomics ([Metabolights](https://www.ebi.ac.uk/metabolights/), [Metabolomics Workbench](https://www.metabolomicsworkbench.org/), and [Massive](https://massive.ucsd.edu/ProteoSAFe/static/massive.jsp)).  Submitting data to these takes a long time and a lot of effort.  So much so, that lab members sometimes only submit a subset of raw data, and often the annotation provided during submission is minimal.  Our lab has also deposited data in different repositories, so our raw data is spread out.

Now that mzXML are stored, organized, and annotated in TraceBase, we could develop some tool that builds a submission for one of these repositories.

Pros:
 - This could provide real functional benefit for users who upload data to TraceBase.  This could incentivize uploading to TraceBase.
 - Would this reduce our requirement for repository-like functions in a public-facing instance?

Cons:
 - Tying a new function to an external system will create a long-term burden (if they update, we may need to update).
 - Time and effort required to build this tool.
 
 Alternatives:
 - Write an external app or script which takes a download of mzXML from TraceBase and builds the submission for MetaboLights (perhaps Michael N or someone else in lab writes this).  If this actually works then we could decide to integrate into tracebase later.

**Comment by mneinast** [comment]
I've surveyed past and present lab members about their experience uploading raw data.  Here is a summary:

- Nobody has uploaded raw files from tracing experiments.  When they do upload raw files, they only upload from "metabolomics" experiments which don't include infusions.
- 50% of surveyed members said they have not uploaded the raw files for their manuscript at all, even though it is a requirement for most journals and funding agencies now. (Most do upload "raw results", meaning the compounds which they have curated, in the form of supplemental xlsx files).  In one case, a former member who performed the LCMS has stopped responding to requests for help in uploading the raw files.
- When people have uploaded raw files, they have not included all required sample information to analyze the data easily.
- Practically, lab members report spending 6-12 hours to upload data from one simple metabolomics experiment.  They often do this during a stressful period late in the publication process.  Clearly this is not practical for tracing experiments.
- Lab members have deposited data to all three repositories listed above.

**Reply by jcmatese** [reply]
Thanks so much for the survey!

**Comment by mneinast** [comment]
Mike Skinnider had this feedback on which repository is ideal:

> Yes: strong preference for MetaboLights > Workbench > MassIVE/GNPS. 
> Advantages of MetaboLights from my perspective as a re-user of data: 
> 1. supports file transfer via Globus, which is much faster than FTP
> 2. requires (or at least promotes) including extensive metadata about the experimental protocols, sample characteristics (species, tissue, disease, etc.), metabolite annotation
> 3. the MetaboLights team curates uploaded data to make sure sample and metabolite tables are in the same format across every study
> This all makes it way easier to figure out what’s going on at a technical or biological level when the data is from MTBLS.

**Comment by mneinast** [comment]
Also, yesterday's discussion with John Wiggins led us to the conclusion that a public-facing version of tracebase would not be a repository.  So instead, this would be an important feature to implement.
